### PR TITLE
feat: ✨ syn-details

### LINKED ISSUE #462: feat: ✨ syn-details
# Story

As a user of the Synergy Design System, I want to have the possibility to show an item of headline/content pairs in a space saving way.

# Acceptance criteria

- [x] All DoD items successfully completed
- [x] All kind of information (text, images, etc.) can be displayed in a `<syn-detail>`.
- [x] `<syn-detail>` accepts a headline and content
- [x] `<syn-detail>` must be accessible
- [x] All possible items must be overridable (e.g. via part or slots)

### LINKED ISSUE #464: feat: ✨ syn-accordion
# Story

As a user of the Synergy Design System, I want to have the possibility to group multiple `<syn-detail>` elements to create an easy to use navigation.

# Acceptance criteria

- [x] All DoD items successfully completed
- [x] Allow to group multiple `<syn-detail>` items
- [x] Implementation according to `<sd-accordion-group>` (https://github.com/solid-design-system/solid/blob/main/packages/components/src/components/accordion-group/accordion-group.ts)

### PR DESCRIPTION
**Pull request by schilchSICKAG**
# Pull Request

## 📖 Description

This PR adds the new `<syn-details>` and `<syn-accordion>` components.

### 🎫 Issues

- Closes #462
- Closes #464

## 👩‍💻 Reviewer Notes

Next to the new components, the following adjustments where made that have to be checked:

- A new size-token `syn-spacing-medium-large` was added. It points to a size of `20px` and sits between the `syn-spacing-medium` (`16px`) and `syn-spacing-large` (`24px`) tokens.
- The documentation tokens originally in branch `origin/feat/change-syn-details-docs` where included into this dev-branch.

## 📑 Test Plan

- As usual compare the stories in Storybook
- Have a look at the code of `<syn-accordion>` and make sure everything works and is accessible.

## ✅ DoD

- [x] I have tested my changes manually.
- [x] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [x] I have added documentation to the DaVinci migration guide (if need be)
- [x] I have made sure to follow the projects coding and contribution guides.
- [x] I have made sure that the bundle size has changed appropriately.
- [x] I have validated that there are no accessibility errors.
- [x] I have used design tokens instead of fix css values

### CODE REVIEW

**Review by kirchsuSICKAG — CHANGES_REQUESTED** [review]

File: packages/components/src/components/details/details.custom.styles.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    font: var(--syn-body-medium-regular);
+  }
+
+  .details--size-small .details__summary::slotted(syn-icon) {
+    font-size: var(--syn-spacing-large);
+    margin-right: var(--syn-spacing-small);
+  }
+
+  .details--size-small .details__summary-icon {
+    font-size: var(--syn-font-size-large);
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/details/details.custom.styles.ts]: I swear it was different yesterday :). Fixed it

File: packages/components/src/components/details/details.custom.styles.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    font: var(--syn-body-large-regular);
+  }
+
+  .details--size-medium .details__summary::slotted(syn-icon) {
+    font-size: var(--syn-spacing-x-large);
+    margin-right: var(--syn-spacing-x-small);
+  }
+
+  .details--size-medium .details__summary-icon {
+    font-size: var(--syn-spacing-x-large);
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/details/details.custom.styles.ts]: I swear it was different yesterday :). Fixed it

File: packages/components/src/components/accordion/accordion.component.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    componentStyles,
+    styles,
+  ];
+
+  @queryAssignedElements({ selector: 'syn-details' }) detailsInDefaultSlot!: SynDetails[];
+
+  /**
+   * Indicates whether or not multiple `<syn-details>` elements can be open at the same time.
+   */
+  @property({ attribute: 'close-others', type: Boolean }) closeOthers = false;
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/accordion/accordion.component.ts:32]: Shoelace does not handle this, too. I also thought about if we should implement this. As we do not know what users want to do, I would still go with the current version. I do not want to come into the way of the user if he opens 5 or 10 of the `<syn-details>` elements and through some change in the code they all close automatically ;)

File: packages/components/src/components/accordion/accordion.component.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        detailsElement.removeAttribute('open');
+      });
+    }
+  };
+
+  render() {
+    /* eslint-disable @typescript-eslint/unbound-method */
+    return html`
+      <div part="">
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/accordion/accordion.component.ts:81]: Error during copying, good catch ;)

File: packages/components/src/components/accordion/accordion.test.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import '../../../dist/synergy.js';
+import { expect, fixture, html } from '@open-wc/testing';
+
+import type SynAccordion from './accordion.js';
+
+describe('<syn-accordion>', () => {
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/accordion/accordion.test.ts:8]: Done

File: packages/components/src/components/accordion/accordion.component.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import type SynDetails from '../details/details.component.js';
+
+/**
+ * @summary Accordions provide the ability to group a list of `<syn-details>`.
+ *
+ * @documentation https://synergy-design-system.github.io/?path=/docs/components-syn-accordion--docs
+ * @status stable
+ * @since 1.23.0
+ *
+ * @slot - The accordion's main content.
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/accordion/accordion.component.ts]: Done

File: packages/components/src/components/accordion/accordion.component.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  ];
+
+  @queryAssignedElements({ selector: 'syn-details' }) detailsInDefaultSlot!: SynDetails[];
+
+  /**
+   * Indicates whether or not multiple `<syn-details>` elements can be open at the same time.
+   */
+  @property({ attribute: 'close-others', type: Boolean }) closeOthers = false;
+
+  /** The size that should be applied to all `<syn-detail>`'s */
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/accordion/accordion.component.ts]: Done

File: packages/components/src/components/accordion/accordion.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kirchsuSICKAG) [reply packages/components/src/components/accordion/accordion.component.ts:32]: Thats fine for me, if we let it as it is  👍

**Comment by klaarseSICKAG** [comment]
Accordion also done! Nice work!
I am not so sure about the default size though. Medium looks so large. I'd rather go for small as default.

**Review by kirchsuSICKAG — APPROVED** [review]

File: packages/components/src/components/details/details.custom.styles.ts [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+   * so we apply the opacity to the inner elements instead
+   */
+  .details--disabled {
+    opacity: 1;
+  }
+
+  .details--disabled summary,
+  .details--disabled .details__body {
+    opacity: var(--syn-opacity-50);
+  }
```
Comment (mariohamann) [inline packages/components/src/components/details/details.custom.styles.ts:21]: @LeaLeRose I'm not sure if this is good. IMO we should stick to our current convention and make the whole element look disabled, including the border at the bottom.

**Comment by mariohamann** [comment]
@kirchsuSICKAG What do you think about taking this one over? As you already got some experience with `sharp` and `contained` this could make sense?

**Comment by klaarseSICKAG** [comment]
please review migration guide: https://gitlab.sick.com/platforms/davinci/synergymigrationguide/-/merge_requests/1

**Review by kirchsuSICKAG — APPROVED** [review]

**Pull request merged**